### PR TITLE
Feature/int 1089

### PR DESCRIPTION
**Pull request by Fibonaccy**
https://jiranewsletter2go.atlassian.net/browse/INT-1089

### CODE REVIEW

File: src/version/107/frontend/nl2go_manager.php [hunk 1]
```diff
@@ -472,9 +472,9 @@ public function setUnsubscribe()
 			self::sendError('EMAILNOTPROVIDED');
 		}
 
-		$GLOBALS["DB"]->executeQuery('DELETE FROM ' . self::NEWSLETTER_SUBSCRIBER_TABLE . " WHERE cEmail = '$email'", 1);
+        $affectedRows = $GLOBALS["DB"]->executeQuery('DELETE FROM ' . self::NEWSLETTER_SUBSCRIBER_TABLE . " WHERE cEmail = '$email'", 3);
```
Comment (Maiky) [inline src/version/107/frontend/nl2go_manager.php]: what is the meaning of this 3?

File: src/version/107/frontend/nl2go_manager.php [hunk 2]
```diff
@@ -521,9 +521,9 @@ public function setSubscribe()
 			now()
 		);";
 
-		$GLOBALS["DB"]->executeQuery($query, 1);
+        $lastInsertId = $GLOBALS["DB"]->executeQuery($query, 7);
```
Comment (Maiky) [inline src/version/107/frontend/nl2go_manager.php]: and what is the meaning of 7 ? 😄  (maybe we should add comments, I'm pretty sure the next one will ask the same question again 😂 )

**Review by Maiky — CHANGES_REQUESTED** [review]

File: src/version/107/frontend/nl2go_manager.php (hunk 1 above)
Comment (Fibonaccy) [reply src/version/107/frontend/nl2go_manager.php]: It determines the return type in this case the number of affected rows will be returned.

File: src/version/107/frontend/nl2go_manager.php (hunk 2 above)
Comment (Fibonaccy) [reply src/version/107/frontend/nl2go_manager.php]: After the query was executed JTLs NiceDB will try to set the return value based on this number. Here is the relevant snippet:

``` php
switch ($return) {
            case 1:
                $ret = $res->fetchObject();
                break;
            case 2:
                $ret = [];
                while (($row = $res->fetchObject()) !== false) {
                    $ret[] = $row;
                }
                break;
            case 3:
                $ret = $res->rowCount();
                break;
            case 7:
                $id = $this->pdo->lastInsertId();
                $ret = ($id > 0) ? $id : 1;
                break;
            case 8:
                $ret = $res->fetchAll(PDO::FETCH_NAMED);
                if (is_array($ret) && isset($ret[0])) {
                    $ret = $ret[0];
                } else {
                    $ret = null;
                }
                break;
            case 9:
                $ret = $res->fetchAll(PDO::FETCH_ASSOC);
                break;
            case 10:
                $ret = $res;
                break;
            case 11:
                $ret = $res->fetchAll(PDO::FETCH_BOTH);
                break;
            default:
                $ret = true;
                break;
        }
```

``` php
   /**
     * 1  - single fetched object
     * 2  - array of fetched objects
     * 3  - affected rows
     * 7  - last inserted id
     * 8  - fetched assoc array
     * 9  - array of fetched assoc arrays
     * 10 - result of querysingle
     */
```

I will add descriptive constants for the returntypes for future use if thats ok with you.

Comment (Maiky) [reply src/version/107/frontend/nl2go_manager.php]: crazy shit oO thanks :)

**Review by Maiky — APPROVED** [review]

**Comment by Maiky** [comment]
I think in preparation for the new version we will need to copy the whole folder from 107 to "108" then, right @mimonewsletter2go ?

**Comment by mimonewsletter2go** [comment]
> I think in preparation for the new version we will need to copy the whole folder from 107 to "108" then, right @mimonewsletter2go ?

Yeah please create a new folder 108 and implement the changes also there. I think we already merged a version 108 of the plugin into the master in this ticket: INT-946

**Pull request merged**